makeflow/src/makeflow.c

```diff
@@ -1523,6 +1538,14 @@ int main(int argc, char *argv[])
 				if(!wrapper) wrapper = makeflow_wrapper_create();
 				makeflow_wrapper_add_output_file(wrapper, optarg);
 				break;
+			case LONG_OPT_SHARED_FS:
+				assert(shared_fs);
+				s = string_combine(xxstrdup(optarg), "/");
+				t = realpath(s, NULL);
+				free(s);
+				if (!t) fatal("can't access shared filesystem %s: %s", optarg, strerror(errno));
+				list_push_head(shared_fs, string_combine(t, "/"));
```

Comment:
Huh, that's odd...
The double append is actually intentional. The first one forces `realpath()` to resolve the path as a directory, which should give back an absolute path without a trailing `/`. The second append is to ensure that shared fs checks work under directories, not just under string prefixes, e.g. should match `/hadoop/foo` but not `/hadoop1`

What OS/libc are you using? Maybe there's some platform weirdness. If that's the case, I'm OK with removing the second append, so that `--shared-fs` matches `/hadoop*`